velox/connectors/hive/iceberg/IcebergPartitionIdGenerator.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/iceberg/IcebergPartitionIdGenerator.h"
+#include "velox/connectors/hive/iceberg/Transforms.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+IcebergPartitionIdGenerator::IcebergPartitionIdGenerator(
+    const RowTypePtr& inputType,
+    std::vector<column_index_t> partitionChannels,
+    uint32_t maxPartitions,
+    memory::MemoryPool* pool,
+    const std::shared_ptr<const IcebergInsertTableHandle>& insertTableHandle,
+    bool partitionPathAsLowerCase)
+    : PartitionIdGenerator(
+          partitionChannels,
+          maxPartitions,
+          partitionPathAsLowerCase),
+      pool_(pool),
+      insertTableHandle_(insertTableHandle) {
+  VELOX_USER_CHECK(pool, "Memory pool cannot be null");
```

Comment:
No need to check the pool, only check it in the IcebergTableHandle or other user exposed location